test/ambiguities.jl

```diff
@@ -1,20 +1,8 @@
-
 # Allow no new ambiguities (see #18), unless you fix some old ones first!
 
-const allowable_ambiguities =
-    if VERSION < v"1.1"
-        3
-    elseif VERSION < v"1.2"
-        1
-    else
-        0
-    end
+const allowable_ambiguities = 0
 
-if VERSION ≥ v"1.6.0"
-    # TODO: Revisit and fix. See
-    #   https://github.com/JuliaLang/julia/pull/36962
-    #   https://github.com/JuliaLang/julia/issues/36951
-    @test_broken length(detect_ambiguities(#=LinearAlgebra, =#StaticArrays)) <= allowable_ambiguities
-else
-    @test length(detect_ambiguities(Base, LinearAlgebra, StaticArrays)) <= allowable_ambiguities
-end
+# TODO: Revisit and fix. See
+#   https://github.com/JuliaLang/julia/pull/36962
+#   https://github.com/JuliaLang/julia/issues/36951
+@test_broken length(detect_ambiguities(#=LinearAlgebra, =#StaticArrays)) <= allowable_ambiguities
```

Comment:
Seems a pity to have this turned off entirely, is there some reasonable nonzero value we can still set this to?

Upstream did seemingly break this test without obvious recourse, so if ignoring this is the best we can do, so be it.

Reply:
Currently we have:
```jl
length(detect_ambiguities(StaticArrays))                # = 49 on v1.6.5 and 48 on v1.7.1
length(detect_ambiguities(LinearAlgebra, StaticArrays)) # = 49 on v1.6.5 and 48 on v1.7.1
```
I'm not sure why the LinearAlgebra thing was commented out.

48 is of course a lot higher than the small numbers that were given for earlier versions before - but I don't understand what made this code and branching on versions necessary in the first place; presumably one of those PRs linked above.

If we want *something* we could set it to e.g. 55 - then we'd at least be told if things suddenly increase a lot. We could also set it lower/higher, depending on tastes?

Reply:
Heh... I rebased this PR just now and that increased the number of ambiguities from 49 to 61 (on v1.6)...
So, yeah, there's some downside to not tracking it I guess.

Reply:
A follow-up on the difference of ambiguity counts. Turns out there's two earlier `@testset`s which define new structs that add to the ambiguity count. Specifically, this happens here:
- 5 ambiguities: https://github.com/JuliaArrays/StaticArrays.jl/blob/d336cfb672a1d5a427b402c8a3e3026cc75d83e3/test/FieldVector.jl#L86-L90
-  7 ambiguities: https://github.com/JuliaArrays/StaticArrays.jl/blob/d336cfb672a1d5a427b402c8a3e3026cc75d83e3/test/FieldMatrix.jl#L94-L98

I.e., the test suite itself adds extra ambiguities: this is probably undesirable, but also an issue that adding certain subtypes of `FieldVector` and `FieldMatrix` introduces new ambiguities.

Reply:
For this PR, I just updated the numbers to `61` (v1.6) and `60` (v1.7). I also made the change that the ambiguity check should be ensure unchanged number ambiguities. If a change decreases the number of ambiguities, it should decrease the values in `/test/ambiguities.jl` accordingly.

Probably, it would be good with some ambiguity hunting at some point...